### PR TITLE
feat: Add box-shadow ui panel for editing values individually for layers

### LINKED ISSUE #2523: Box Shadow UI
Screenshot 2023-10-24 at 11 10 28 AM

The Box Shadow docs have been updated. There is now a design for the menu so users can adjust their box shadows without code. Please read the new docs carefully to notice all the changes.

Most important: "The code and the UI should be synced. This will be very valuable for the user. If valid code is pasted, the UI should change to reflect the pasted values. If the UI values are changed, the code text should change to reflect the UI values."

Figma: https://www.figma.com/file/xCBegXEWxROLqA1Y31z2Xo/%F0%9F%93%96-Webstudio-Design-Docs?type=design&node-id=4512-49938&mode=design

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
## Design
https://www.figma.com/file/xCBegXEWxROLqA1Y31z2Xo/%F0%9F%93%96-Webstudio-Design-Docs?node-id=8075%3A38024&mode=dev

## Description
Box-shadow floating panel to edit each layer values individually. 
fixes #2523 

box-shadow

### CODE REVIEW

File: apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx [hunk 1]
```diff
@@ -1,33 +1,110 @@
...
+import { useMemo, useState } from "react";
 import type { IntermediateStyleValue } from "../../shared/css-value-input";
+import { CssValueInputContainer } from "../../controls/position/css-value-input-container";
+import { toPascalCase } from "../../shared/keyword-utils";
+import { ColorControl } from "../../controls";
+import type { SetProperty } from "../../shared/use-style-data";
+
+/*
+  We can't split and use individual css properties for box-shadow.
+  So, the best way to get eh validation from inputs is to use a fallback
```
Comment (istarkov) [inline apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx:42]: what's `eh`?

File: apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx [hunk 2]
```diff
@@ -1,33 +1,110 @@
...
+  css-property that accepts the same value as the input that we are trying to validate.
+
+  box-shadow: color, inset, offsetX, offsetY, blur, spread;
+
+  - offsetX -> length -> +ve & -ve
```
Comment (istarkov) [inline apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx:47]: what means `+ve & -ve` ?

File: apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx [hunk 3]
```diff
@@ -49,51 +126,206 @@ export const BoxShadowContent = (props: BoxShadowContentProps) => {
...
+        css={{
+          paddingTop: theme.spacing[5],
+          px: theme.spacing[9],
+        }}
+      >
+        <Flex direction="column">
+          <Label css={{ display: "inline" }}>X</Label>
+          <CssValueInputContainer
+            key="boxShadowOffsetX"
+            property="outlineOffset"
```
Comment (istarkov) [inline apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx:163]: Probably comment that outlineOffset is a "fake" property would be good to write in comment here pointing to the very good comment above

File: packages/icons/icons/shadow-inset.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M1.64998 6.99998C1.64998 4.04525 4.04525 1.64998 6.99998 1.64998C9.9547 1.64998 12.35 4.04525 12.35 6.99998C12.35 9.9547 9.9547 12.35 6.99998 12.35C4.04525 12.35 1.64998 9.9547 1.64998 6.99998ZM6.99998 0.349976C3.32728 0.349976 0.349976 3.32728 0.349976 6.99998C0.349976 10.6727 3.32728 13.65 6.99998 13.65C10.6727 13.65 13.65 10.6727 13.65 6.99998C13.65 3.32728 10.6727 0.349976 6.99998 0.349976ZM4.31501 7.00001C4.31501 5.51713 5.51713 4.31501 7.00001 4.31501C8.4829 4.31501 9.68501 5.51713 9.68501 7.00001C9.68501 7.359 9.97603 7.65001 10.335 7.65001C10.694 7.65001 10.985 7.359 10.985 7.00001C10.985 4.79916 9.20087 3.01501 7.00001 3.01501C4.79916 3.01501 3.01501 4.79916 3.01501 7.00001C3.01501 7.359 3.30603 7.65001 3.66501 7.65001C4.024 7.65001 4.31501 7.359 4.31501 7.00001Z" fill="#11181C"/>
```
Comment (istarkov) [inline packages/icons/icons/shadow-inset.svg:2]: See how every other svg is made. `fill` is removed from path and set as `currentColor`  at `svg` element level
Plz fix and regenerate

**Review by istarkov — APPROVED** [review]
Good work! Fix svg plz, and alittle bit improve comments.

File: apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx (hunk 1 above)
Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx:42]: Just a typo, going to correct the comment 👍 😅

File: apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx (hunk 2 above)
Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx:47]: offsets and spread, takes both positive and negative values. But, the blur takes only a positive value. I will write it explicitly there 👍

**Review by taylornowotny — COMMENTED** [review]
Thanks @JayaKrishnaNamburu ! You've been a great help.

I will not be able to test as I am traveling to see family for the holidays. After that I might not be an active part of Webstudio any more.

Your design point of contact is now @markxtang

**Comment by JayaKrishnaNamburu** [comment]
@taylornowotny it was nice interacting with you too. Thanks for letting me know 😄. Happy holidays 🌲

**Comment by markxtang** [comment]
@JayaKrishnaNamburu Amazing work! The demo looks so good!!

Only thing I didn't see shown in the demo is the ability to change **px** to a different unit. I realize we didn't specify that in our design. I updated the docs page to make that clear. They're the same units as the width/height without the percentage. I'm open to discuss if you want to include/remove certain units. 

Here's the new section in the docs.
https://www.figma.com/file/xCBegXEWxROLqA1Y31z2Xo/%F0%9F%93%96-Webstudio-Design-Docs?type=design&node-id=9476%3A59835&mode=dev
image

For the placement of the menu, you can refer to width/height inputs in the Size Section in the Style Panel.
image

**Comment by JayaKrishnaNamburu** [comment]
Screenshot 2023-12-21 at 00 49 32
@markxtang you can change them. 

And we can't add`%` to this, if you are referring to it. Since, `%` are invalid for `box-shadows` from the css spec 👍

**Comment by markxtang** [comment]
perfect!

**Review by markxtang — APPROVED** [review]

**Pull request merged**